src/redux/reducers/TodoReducer.ts

```diff
@@ -3,18 +3,19 @@ import { AddItemPayload } from '../actions/todoActions/addItemAction';
 import { ItemDeletePayload } from '../actions/todoActions/deleteItemAction';
 import { TodoActionTypes } from '../actions/todoActions';
 import { DoneItemPayload } from '../actions/todoActions/doneItemAction';
-import { SignInPayload } from '../actions/todoActions/signInAction';
+import { AuthActionsTypes } from '../actions/authActions';
+import { SignInPayload } from '../actions/authActions/signInAction';
 
 export type TodoReducerState = {
   todoItems: TodoItemType[];
   userToken: SignInPayload['userToken'] | null;
```

Comment:
you should use payload type for Reducer type

reducer has more common type so you can do vice versa

Reply:
I do not understand

Reply:
you **shouldn't** use payload type for Reducer type

your payload can have Reducer type
for example

PayloadType = {
id: ReducerType['user']['id']
}

Reply:
i fix it